### PR TITLE
#169392375 refactor accommodations

### PR DESCRIPTION
**Pull request by alexismajya**
#### What does this PR do?

- refactor accommodations controllers and related files

#### Description of Task to be completed?

- refactor accommodation controller
- create accommodation services
- refactor accommodation route

#### How should this be manually tested?

- clone this branch

#### What are the relevant pivotal tracker stories?

[Pivotal Tracker story](https://www.pivotaltracker.com/story/show/169392375)

#### Questions:

### CODE REVIEW

File: src/controllers/accommodations.controller.js [hunk 1]
```diff
@@ -17,10 +17,7 @@ const uploadImages = (req, res, data, msg) => {
     cloudinary.uploader.upload(element.tempFilePath, async (result, error) => {
       if (error) { failure += 1; checkIfAllUploaded(numberofImages, res, data, msg); }
 
-      const imageRes = await db.AccommodationImages.create({
-        imageurl: result.url,
-        accommodationid: data.id,
-      });
+      const imageRes = await accommodationService.createAccommodationImage(result.url, data.id,);
```
[the remark this answers is not in the document]
Comment (alexismajya) [reply src/controllers/accommodations.controller.js]: ok

File: src/services/accommodation.service.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    const accommodations = await db.Accommodations.findAll({
+      where: { userid: hostId },
+      include: [{ model: db.AccommodationImages, attributes: { exclude: ['id', 'accommodationid', 'createdAt', 'updatedAt'] } }],
+    });
+    return accommodations;
+  },
+
+  async createAccommodation(
+    name, description, address, availableSpace, cost, services, amenities, userId
+  ) {
```
[the remark this answers is not in the document]
Comment (alexismajya) [reply src/services/accommodation.service.js]: resolved

File: src/services/accommodation.service.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  async getAllAccommodationsByHost(hostId) {
+    const accommodations = await db.Accommodations.findAll({
+      where: { userid: hostId },
+      include: [{ model: db.AccommodationImages, attributes: { exclude: ['id', 'accommodationid', 'createdAt', 'updatedAt'] } }],
+    });
+    return accommodations;
+  },
+
+  async createAccommodation(
```
[the remark this answers is not in the document]
Comment (alexismajya) [reply src/services/accommodation.service.js]: resolved

File: src/controllers/accommodations.controller.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import sendResult from '../utils/sendResult';
-import db from '../database/models/index';
+import accommodationService from '../services/accommodation.service';
```
Comment (elchroy) [inline src/controllers/accommodations.controller.js]: Let's use `AccommodationService` here

File: src/routes/accommodationRoute.js [hunk 5]
```diff
@@ -1,11 +1,11 @@
 import express from 'express';
 import fileUpload from 'express-fileupload';
 import path from 'path';
-import accommodationCont from '../controllers/accommodationController';
-import accMidd from '../middlewares/accommodationMiddleware';
-import userMidd from '../middlewares/userMiddlware';
-import valid from '../validation';
-import roles from '../middlewares/rolesMiddlewares';
+import accommodationsController from '../controllers/accommodations.controller';
```
Comment (elchroy) [inline src/routes/accommodationRoute.js]: Let's use `AccommodationService` here, as well as `AccommodationMiddlware` and `AccommodationsController`

**Review by elchroy — CHANGES_REQUESTED** [review]
- Let's use `PascalCase` for controller, service and middleware names. Controller names should also be pluralized.
- `AccommodationService`, `AccommodationMiddlware` and `AccommodationsController`

File: src/services/accommodation.service.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import db from '../database/models/index';
```
Comment (elchroy) [inline src/services/accommodation.service.js:1]: default export should be a const named `AccommodationService`

File: src/routes/accommodationRoute.js (hunk 5 above)
Comment (alexismajya) [reply src/routes/accommodationRoute.js]: resolved

**Comment by alexismajya** [comment]
> * Let's use `PascalCase` for controller, service and middleware names. Controller names should also be pluralized.
> * `AccommodationService`, `AccommodationMiddlware` and `AccommodationsController`

I applied those changes

File: src/services/accommodation.service.js (hunk 6 above)
Comment (alexismajya) [reply src/services/accommodation.service.js:1]: I applied the changes

File: src/services/accommodation.service.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    });
+    return accommodations;
+  },
+
+  async createAccommodation(request, userId) {
+    const {
+      name, description, address, availableSpace, cost, services, amenities
+    } = request;
+    const newAccommodation = await db.Accommodations.create({
+      name, description, address, availableSpace, cost, services, amenities, userid: userId
```
Comment (lemoissonM) [inline src/services/accommodation.service.js:30]: you can replace all this line by : {...requst, userid: userId} and avoid destructuring the request variable

**Review by lemoissonM — CHANGES_REQUESTED** [review]

**Pull request merged**